components/header/tests/unit/header.spec.js

```diff
@@ -1,3 +1,188 @@
-test('TODO replace this', () => {
-	expect(true).toBe(true);
+import React from 'react';
+import { Header } from '@westpac/header';
+import { Button } from '@westpac/button';
+import { render, within } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { GEL } from '@westpac/core';
+import wbc from '@westpac/wbc';
+
+// The default tests every component should run
+overridesTest({
+	name: 'header', // the name has to be the package name without '@westpac/' scope
+	overrides: ['Header', 'Inner', 'Left', 'LeftBtn', 'LogoLink', 'Logo', 'Right'], // every single override root key
+	Component: (props) => (
+		<Header leftIcon="arrow" {...props}>
+			Header content
+		</Header>
+	), // the component with all components rendered
+});
+
+// another default test to check that the component errors when outside of GEL and renders when inside
+nestingTest({
+	name: 'header',
+	Component: (props) => <Header {...props} />,
+});
+
+describe('Header component', () => {
+	const originalError = console.error;
+	beforeEach(() => {
+		console.error = jest.fn();
+	});
+	afterEach(() => {
+		console.error = originalError;
+	});
+
+	test('should render its content', () => {
+		const text = 'Hello';
+		const content = <span data-testid="my-span">{text}</span>;
+		const SimpleHeader = () => (
+			<GEL brand={wbc}>
+				<Header>{content}</Header>
+			</GEL>
+		);
```

Comment:
minor(reafactor): I think this component could be outside and reused across the tests since it has been repeated.

Reply:
Yes, ok, thanks for the tip; made that change. Please take a squiz.